modules/item/src/main/java/unnamed/mmo/item/crafting/CraftingInventory.java

```diff
@@ -0,0 +1,197 @@
+package unnamed.mmo.item.crafting;
+
+import net.kyori.adventure.text.Component;
+import net.minestom.server.entity.Player;
+import net.minestom.server.inventory.Inventory;
+import net.minestom.server.inventory.InventoryType;
+import net.minestom.server.inventory.PlayerInventory;
+import net.minestom.server.inventory.TransactionOption;
+import net.minestom.server.inventory.click.InventoryClickResult;
+import net.minestom.server.item.ItemStack;
+import net.minestom.server.utils.inventory.PlayerInventoryUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.TestOnly;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class CraftingInventory extends Inventory {
+
+    private final RecipeList recipeList;
+
+    // The crafting menu indices are laid out as follows
+    // 1 2 3
+    // 4 5 6 -> 0
+    // 7 8 9
+    public CraftingInventory(RecipeList list) {
+        super(InventoryType.CRAFTING, Component.text("Ultimate Supreme Crafting Menu"));
+        this.recipeList = list;
+    }
+
+    @Override
+    public boolean leftClick(@NotNull Player player, int slot) {
+        if (slot == 0) {
+            // Deny placements into crafting slot
+            if (!getCursorItem(player).isAir()) {
```

Comment:
if they are the same item and can stack according to their stacking rule this should be allowed (think clicking many times to craft sticks over and over)
+tests for this!

Reply:
I see that this is done for right click now. Vanilla supports both so we should too